src/Microsoft.DotNet.Configurer/NuGetCachePrimer.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Configurer
+{
+    public class NuGetCachePrimer : INuGetCachePrimer
+    {
+        private const string NUGET_SOURCE_PARAMETER = "-s";
+        private readonly ICommandFactory _commandFactory;
+        private readonly IDirectory _directory;
+        private readonly IFile _file;
+        private readonly INuGetPackagesArchiver _nugetPackagesArchiver;
+        private readonly INuGetCacheSentinel _nuGetCacheSentinel;
+
+        public NuGetCachePrimer(
+            ICommandFactory commandFactory,
+            INuGetPackagesArchiver nugetPackagesArchiver,
+            INuGetCacheSentinel nuGetCacheSentinel)
+            : this(commandFactory,
+                nugetPackagesArchiver,
+                nuGetCacheSentinel,
+                FileSystemWrapper.Default.Directory,
+                FileSystemWrapper.Default.File)
+        {
+        }
+
+        internal NuGetCachePrimer(
+            ICommandFactory commandFactory,
+            INuGetPackagesArchiver nugetPackagesArchiver,
+            INuGetCacheSentinel nuGetCacheSentinel,
+            IDirectory directory,
+            IFile file)
+        {
+            _commandFactory = commandFactory;
+            _directory = directory;
+            _nugetPackagesArchiver = nugetPackagesArchiver;
+            _nuGetCacheSentinel = nuGetCacheSentinel;
+            _file = file;
+        }
+
+        public void PrimeCache()
+        {
+            if(SkipPrimingTheCache())
+            {
+                return;
+            }
+
+            var pathToPackagesArchive = _nugetPackagesArchiver.ExtractArchive();
+
+            PrimeCacheUsingArchive(pathToPackagesArchive);
+        }
+
+        private bool SkipPrimingTheCache()
+        {
+            return !_file.Exists(_nugetPackagesArchiver.NuGetPackagesArchive);
```

Comment:
I just saw the test. I'm worried that we are ignoring potential a potential error scenario. For example if this happened in stage2 would anything throw? 

I'm okay with leaving the method but can we throw in this scenario and expect the failure in the test?

---
In reply to: [66658329](https://github.com/dotnet/cli/pull/3485#discussion_r66658329) [](ancestors = 66658329)